src/main/java/com/coveros/selenified/element/check/wait/WaitForEquals.java

```diff
@@ -0,0 +1,568 @@
+        double end = System.currentTimeMillis() + (seconds * 1000);
+        try {
+            elementPresent(seconds);
+            while (!element.get().tableCell(row, col).get().text().equals(expectedText) && System.currentTimeMillis() < end) ;
+            double timeTook = Math.min((seconds * 1000) - (end - System.currentTimeMillis()), seconds * 1000) / 1000;
+            checkText(row, col, expectedText, seconds, timeTook);
+            if (!element.get().tableCell(row, col).get().text().equals(expectedText)) {
+                file.addError();
+            }
+        } catch (TimeoutException e) {
+            checkText(row, col, expectedText, seconds, seconds);
+            file.addError();
+        }
+    }
+
+    /**
+     * Waits for the element's value equals the provided expected value. If
+     * the element isn't present or an input, this will constitute a failure,
+     * same as a mismatch.
+     * The provided wait time will be used and if the element doesn't
+     * have the desired match count at that time, it will fail, and log
+     * the issue with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue - the expected input value of the element
+     * @param seconds       - how many seconds to wait for
+     */
+    public void value(String expectedValue, double seconds) {
+        double end = System.currentTimeMillis() + (seconds * 1000);
+        try {
+            double timeTook = elementPresent(seconds);
+            WebDriverWait wait = new WebDriverWait(element.getDriver(), (long) (seconds - timeTook), DEFAULT_POLLING_INTERVAL);
+            wait.until(ExpectedConditions.textToBePresentInElementValue(element.defineByElement(), expectedValue));
+            timeTook = Math.min((seconds * 1000) - (end - System.currentTimeMillis()), seconds * 1000) / 1000;
+            checkValue(expectedValue, seconds, timeTook);
+        } catch (TimeoutException e) {
+            checkValue(expectedValue, seconds, seconds);
+            file.addError();
+        }
+    }
+
+    /**
+     * Waits for the element's selected option equals the provided expected
+     * option. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch.
+     * The provided wait time will be used and if the element doesn't
+     * have the desired match count at that time, it will fail, and log
+     * the issue with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedText - the expected input text of the element
+     * @param seconds      - how many seconds to wait for
+     */
+    public void selectedOption(String expectedText, double seconds) {
+        double end = System.currentTimeMillis() + (seconds * 1000);
+        try {
+            elementPresent(seconds);
+            if( !element.is().select()) {
+                throw new TimeoutException("Not a select");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Not a select" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)